packages/thirdweb/src/react/native/ui/connect/ConnectButton.tsx

```diff
@@ -0,0 +1,147 @@
+import { useEffect, useRef, useState } from "react";
+import {
+  Animated,
+  Dimensions,
+  Easing,
+  KeyboardAvoidingView,
+  Modal,
+  Pressable,
+  StyleSheet,
+  View,
+} from "react-native";
+import type { MultiStepAuthProviderType } from "../../../../wallets/in-app/core/authentication/type.js";
+import type { Wallet } from "../../../../wallets/interfaces/wallet.js";
+import { parseTheme } from "../../../core/design-system/CustomThemeProvider.js";
+import type { ConnectButtonProps } from "../../../core/hooks/connection/ConnectButtonProps.js";
+import { useActiveAccount } from "../../hooks/wallets/useActiveAccount.js";
+import { useActiveWallet } from "../../hooks/wallets/useActiveWallet.js";
+import { ThemedButton } from "../components/button.js";
+import { ThemedText } from "../components/text.js";
+import { ConnectModal } from "./ConnectModal.js";
+import { ConnectedButton } from "./ConnectedButton.js";
+
+export type ModalState =
+  | { screen: "base" }
+  | { screen: "otp"; auth: MultiStepAuthProviderType; wallet: Wallet<"inApp"> }
+  | { screen: "external_wallets" };
+
+export function ConnectButton(props: ConnectButtonProps) {
```

Comment:
Does this need jsdocs?

Reply:
Yes, even though most ppl will import via /react so will get the tsdoc of react.

I'll still document this tho for ppl importing from /react-native

Reply:
added in part 2 PR